src/components/modals/RequestFioAddressModal.js

```diff
@@ -0,0 +1,222 @@
+import type { EdgeCurrencyConfig } from 'edge-core-js'
+import React, { Component } from 'react'
+import { ActivityIndicator, Clipboard, Image, Text, View } from 'react-native'
+import { sprintf } from 'sprintf-js'
+
+import fioRequestsIcon from '../../assets/images/fio/SendModule_FioAddress.png'
+import s from '../../locales/strings.js'
+import styles from '../../styles/scenes/ScaneStyle'
+
+type AddressModalProps = {
+  onDone: ({ fioAddress: string, memo: string } | null) => mixed,
+  fioPlugin: EdgeCurrencyConfig,
+  isConnected: boolean
+}
+
+type AddressModalState = {
+  clipboard: string,
+  address: string,
+  addressError: string,
+  memo: string,
+  memoError: string,
+  addressFocus: boolean,
+  memoFocus: boolean,
+  addressValidationLoading: boolean
+}
+export class FioAddressModal extends Component<AddressModalProps, AddressModalState> {
+  constructor (props: AddressModalProps) {
+    super(props)
+    this.state = {
+      clipboard: '',
+      address: '',
+      addressError: '',
+      memo: '',
+      memoError: '',
+      addressFocus: true,
+      memoFocus: false,
+      addressValidationLoading: false
+    }
+  }
+
+  componentDidMount () {
+    this._setClipboard()
+  }
+
+  _setClipboard = async () => {
+    try {
+      const address = await Clipboard.getString()
+      const isFioAddress = await this.props.fioPlugin.otherMethods.isFioAddressValid(address)
+      if (!isFioAddress) return
+
+      this.setState({
+        clipboard: address
+      })
+    } catch (e) {
+      //
+    }
+  }
+
+  updateAddress = (address: string) => {
+    this.setState({
+      address
+    })
+  }
+
+  updateMemo = (memo: string) => {
+    this.setState({
+      memo
+    })
+  }
+
+  onPasteFromClipboard = () => {
+    const { clipboard } = this.state
+    this.updateAddress(clipboard)
+  }
+
+  onCancel = () => {
+    this.props.onDone(null)
+  }
+
+  onSubmit = async () => {
+    const validAddress = await this.validateAddress()
+    const validMemo = this.validateMemo()
+    if (validAddress && validMemo) {
+      this.props.onDone({ fioAddress: this.state.address, memo: this.state.memo })
+    }
+  }
+
+  onValidateAddress = (): void => {
+    this.validateAddress()
+  }
+
+  onValidateMemo = (): void => {
+    this.validateMemo()
+  }
+
+  validateAddress = async (): Promise<boolean> => {
+    const address: string = this.state.address
+    let accountIsAvailable
+    this.setState({
+      addressValidationLoading: true
+    })
+    try {
+      accountIsAvailable = await this.props.fioPlugin.otherMethods.isAccountAvailable(address)
+    } catch (e) {
+      accountIsAvailable = false
+    }
+    if (address && accountIsAvailable) {
+      this.setState({
+        addressError: '',
+        addressFocus: false,
+        memoFocus: true,
+        addressValidationLoading: false
+      })
+      return true
+    } else {
+      this.setState({
+        addressError: s.strings.fragment_send_send_to_fio_error_inline,
```

Comment:
Rename to

s.strings.send_to_fio_error_addr_not_available

Reply:
renamed to `send_fio_request_error_addr_not_available`